Add more smart pointers to AXCoreObject.cpp

abigailfox opened the pull request:
#### 9e77b37ecd015142c1ec3de9131c1b262a7f1b0e
<pre>
Add more smart pointers to AXCoreObject.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=272422">https://bugs.webkit.org/show_bug.cgi?id=272422</a>
<a href="https://rdar.apple.com/problem/126168150">rdar://problem/126168150</a>

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::hasPopup const):
(WebCore::AXCoreObject::tableLevel const):
(WebCore::AXCoreObject::selectedCells):
(WebCore::AXCoreObject::linkedObjects const):
(WebCore::AXCoreObject::appendRadioButtonGroupMembers const):

Canonical link: <a href="https://commits.webkit.org/277337@main">https://commits.webkit.org/277337@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e184ecc45bf6e4729616f5fd1ceca3c81aedd8e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47377 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26557 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50031 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50060 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43425 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49684 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32075 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24017 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/38572 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47958 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/24087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40823 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19892 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21525 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41989 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5420 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/43717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/42389 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51936 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/22407 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/18745 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23682 "Built successfully") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/44902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10437 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/24468 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23401 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/accessibility/AXCoreObject.cpp:
```diff
@@ -566,7 +566,7 @@ void AXCoreObject::appendRadioButtonGroupMembers(AccessibilityChildrenVector& li
+        for (RefPtr parent = parentObject(); parent; parent = parent->parentObject()) {
             if (parent->roleValue() == AccessibilityRole::RadioGroup) {
                 appendRadioButtonDescendants(*parent, linkedUIElements);
                 break;
```
There is an additional `auto*` on line 584. Does that need to be updated too?

abigailfox replied:
`layoutContext()` and `isInRenderTreeLayout()` are both inline getters/checks, so I didn't think it met our threshold for converting, but I'll check in on it

stwrt replied:
@abigailfox So do inline getters/checks not meet the threshold for converting? This issue was not resolved before trying to land. It would be helpful to know for future reference.

cdumez replied:
@stwrt Abigail is following our smart pointer guidelines, which are well documented and were shared with the rest of the team.

stwrt replied:
> @stwrt Abigail is following our smart pointer guidelines, which are well documented and were shared with the rest of the team.

Thanks for the confirmation Chris.

cdumez replied:
Looks like I forgot to put a link: https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines
The beginning of the document is about smart pointers and explains the "trivial functions" aspect.

abigailfox committed:
Add more smart pointers to AXCoreObject.cpp
https://bugs.webkit.org/show_bug.cgi?id=272422
rdar://problem/126168150

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::hasPopup const):
(WebCore::AXCoreObject::tableLevel const):
(WebCore::AXCoreObject::selectedCells):
(WebCore::AXCoreObject::linkedObjects const):
(WebCore::AXCoreObject::appendRadioButtonGroupMembers const):

Canonical link: https://commits.webkit.org/277337@main
